plugins/woocommerce/client/admin/client/core-profiler/index.tsx

```diff
@@ -473,26 +469,23 @@ const updateBusinessInfo = fromPromise(
 			payload: BusinessInfoPayload;
 		};
 	} ) => {
-		const refreshedOnboardingProfile = ( await resolveSelect(
-			OPTIONS_STORE_NAME
-		).getOption( 'woocommerce_onboarding_profile' ) ) as OnboardingProfile;
-
-		await updateStoreCurrency( input.payload.storeLocation );
-
-		return dispatch( OPTIONS_STORE_NAME ).updateOptions( {
-			blogname: input.payload.storeName,
-			woocommerce_default_country: input.payload.storeLocation,
-			woocommerce_onboarding_profile: {
-				...refreshedOnboardingProfile,
+		return Promise.all( [
+			updateStoreCurrency( input.payload.storeLocation ),
+			dispatch( ONBOARDING_STORE_NAME ).updateProfileItems( {
 				is_store_country_set: true,
-				industry: [ input.payload.industry ],
 				is_agree_marketing: input.payload.isOptInMarketing,
-				store_email:
-					input.payload.storeEmailAddress.length > 0
-						? input.payload.storeEmailAddress
-						: null,
-			},
-		} );
+				...( input.payload.industry && {
+					industry: [ input.payload.industry ],
+				} ),
+				...( input.payload.storeEmailAddress.length > 0 && {
+					store_email: input.payload.storeEmailAddress,
```

Comment:
I realized that merchants can't remove email address by emptying the field. It'll still populated by the previously saved value. I'm unsure what's the intended behaviour, but I think there would be cases where merchant wants to empty the email field. What do you think?

Reply:
that’s a very good catch, I’ll make the API nullable instead then